ssr/src/utils/device_id.rs

```diff
@@ -0,0 +1,40 @@
+use std::env;
+
+use leptos::{expect_context, server, ServerFnError};
+
+#[cfg(feature = "ga4")]
+#[server]
+pub async fn send_principal_and_token_offchain(
+    device_id: String,
+    principal_id: String,
+) -> Result<(), ServerFnError> {
+    use crate::utils::off_chain;
+    use tonic::metadata::MetadataValue;
+    use tonic::transport::Channel;
+    use tonic::Request;
+
+    let channel: Channel = expect_context();
+
+    let mut off_chain_agent_grpc_auth_token = env::var("GRPC_AUTH_TOKEN").expect("GRPC_AUTH_TOKEN");
```

Comment:
Is authentication required for this? @komal-rs

Reply:
Not sure if this use case needs. But since it is added behind OffChainService proto, it needs as a result.